packages/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts

```diff
@@ -0,0 +1,239 @@
+          state.namePrefix = 'PolicyAPIGWAuth';
+          this.createPoliciesFromResources(state);
+          ({ roleCount: authRoleCount, policyDocSize: authPolicyDocSize, managedPolicy: authManagedPolicy } = state);
+        }
+
+        if (apiGateway.params.privacy.unauth) {
+          state.methods = path?.privacy?.unauth ?? [];
+          state.roleCount = unauthRoleCount;
+          state.roleName = unauthRoleName;
+          state.policyDocSize = unauthPolicyDocSize;
+          state.managedPolicy = unauthManagedPolicy;
+          state.namePrefix = 'PolicyAPIGWUnauth';
+          this.createPoliciesFromResources(state);
+          ({ roleCount: unauthRoleCount, policyDocSize: unauthPolicyDocSize, managedPolicy: unauthManagedPolicy } = state);
+        }
+      });
+    });
+  }
+
+  toCloudFormation() {
+    prepareApp(this);
+    return this._toCloudFormation();
+  }
+
+  private createPoliciesFromResources(options: ApiGatewayPolicyCreationState) {
+    const { apiGateway, apiRef, env, roleName, path, methods, namePrefix } = options;
+
+    methods.forEach(method => {
+      const policySizeIncrease = computePolicySizeIncrease(method.length, path.policyResourceName.length, apiGateway.resourceName.length);
+
+      options.policyDocSize += policySizeIncrease;
+      // If a managed policy hasn't been created yet, or the maximum
+      // managed policy size has been exceeded, then create a new policy.
+      if (options.roleCount === 0 || options.policyDocSize > MAX_MANAGED_POLICY_SIZE) {
+        // Initial size of 100 for version, statement, etc.
+        options.policyDocSize = 100 + policySizeIncrease;
+        options.roleCount++;
+        options.managedPolicy = createManagedPolicy(this, `${namePrefix}${options.roleCount}`, (roleName as unknown) as string);
+      }
+
+      options.managedPolicy.policyDocument.Statement[0].Resource.push(
+        createApiResource(this.region, this.account, apiRef, env, method, `${path.policyResourceName}/*`),
+        createApiResource(this.region, this.account, apiRef, env, method, path.policyResourceName),
+      );
+    });
+  }
+}
+
+function createManagedPolicy(stack: cdk.Stack, policyName: string, roleName: string): iam.CfnManagedPolicy {
+  return new iam.CfnManagedPolicy(stack, policyName, {
+    roles: [roleName],
+    managedPolicyName: policyName,
+    policyDocument: {
+      Version: '2012-10-17',
+      Statement: [{ Effect: 'Allow', Action: ['execute-api:Invoke'], Resource: [] }],
+    },
+  });
+}
+
+function createApiResource(region, account, api, env, method, resourceName) {
+  return cdk.Fn.join('', [
+    'arn:aws:execute-api:',
+    region,
+    ':',
+    account,
+    ':',
+    api,
+    '/',
+    (cdk.Fn.conditionIf('ShouldNotCreateEnvResources', 'Prod', env) as unknown) as string,
+    method,
+    resourceName,
+  ]);
+}
+
+function computePolicySizeIncrease(methodLength: number, pathLength: number, nameLength: number): number {
```

Comment:
does this compute an increase, or just the new policy size?

Reply:
It computes the increase, and is added to the current policy size on line 122.